source/portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c

```diff
@@ -207,32 +318,62 @@ BaseType_t xNetworkInterfaceInitialise( void )
                 #endif
             }
 
-            /* Initialize the mac and set the MAC address. */
-            XEmacPs_SetMacAddress( pxEMAC_PS, ( void * ) ipLOCAL_MAC_ADDRESS, 1 );
+            /* Initialize the mac and set the MAC address at position 1. */
+            XEmacPs_SetMacAddress( pxEMAC_PS, ( void * ) pxEndPoint->xMACAddress.ucBytes, 1 );
 
             #if ( ipconfigUSE_LLMNR == 1 )
             {
-                /* Also add LLMNR multicast MAC address. */
-                XEmacPs_SetMacAddress( pxEMAC_PS, ( void * ) xLLMNR_MACAddress, 2 );
+                #if ( ipconfigUSE_IPv6 == 0 )
+                {
+                    /* Also add LLMNR multicast MAC address. */
+                    XEmacPs_SetMacAddress( pxEMAC_PS, ( void * ) xLLMNR_MacAddress.ucBytes, 2 );
+                }
+                #else
+                {
+                    /* is this loop really required? */
+                    for( pxEndPoint = FreeRTOS_FirstEndPoint( pxInterface );
```

Comment:
Is it really necessary to add `ucMACAddress` to all endpoints?

I would have thought `XEmacPs_SetHash( pxEMAC_PS, ( void * ) xLLMNR_MacAddressIPv6.ucBytes );` would be sufficient.